tests/webTables.spec.ts

```diff
@@ -0,0 +1,182 @@
+    }
+
+    expect(actualPetsList.sort()).toEqual(expectedPetsList.sort())
+})
+test('validate specialty update', async ({ page }) => {
+    //1. Select the VETERINARIANS menu item in the navigation bar, then select "All"
+    await page.getByText("Veterinarians").click()
+    await page.getByText('All').click()
+
+    //2. On the Veterinarians page, add the assertion that "Rafael Ortega" has specialty "surgery"
+    await expect(page.getByRole('row', { name: 'Rafael Ortega' }).getByRole('cell').nth(1)).toHaveText('surgery')
+
+    //3. Select the SPECIALTIES menu item in the navigation bar
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    //4. Add assertion of the "Specialties" header displayed above the table
+    await expect(page.locator("h2")).toHaveText("Specialties");
+
+    //5. Click on "Edit" button for the "surgery" specialty
+    await page.waitForSelector('tbody')
+
+    const allRows = await page.locator('tbody tr').all()
+    // Iterate through rows to find 'surgery'
+    for (let row of allRows) {
+        if (await row.locator('input').inputValue() === 'surgery') {
+            // ... click 'Edit'
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
+        }
+    }
+    //6. Add assertion "Edit Specialty" page is displayed
+    await expect(page.locator('h2')).toHaveText('Edit Specialty')
+    //7. Update the specialty from "surgery" to "dermatology" and click "Update button"
+    //8. Add assertion that "surgery" was changed to "dermatology" in the list of specialties
+    const specialtyInput = page.locator('#name')
+    await expect(specialtyInput).toHaveValue('surgery')
+    await specialtyInput.clear()
+    await specialtyInput.fill('dermatology')
+    await page.getByText('Update').click()
+    //9. Select the VETERINARIANS menu item in the navigation bar, then select "All"
+    await page.getByText("Veterinarians").click()
+    await page.getByText("All").click()
+    //10. On the Veterinarians page, add assertion that "Rafael Ortega" has specialty "dermatology"
+    await expect(page.getByRole('row', { name: 'Rafael Ortega' }).getByRole('cell').nth(1)).toHaveText('dermatology')
+    //11. Navigate to SPECIALTIES page, revert the changes renaming "dermatology" back to "surgery"
+    await page.getByRole('link', { name: 'Specialties' }).click()
+
+    for (let row of allRows) {
+        if (await row.locator('input').inputValue() === 'dermatology') {
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
+        }
+    }
+
+    await expect(specialtyInput).toHaveValue('dermatology')
+    await specialtyInput.clear()
+    await specialtyInput.fill('surgery')
+    await page.getByText('Update').click()
+})
+
+test('validate specialty lists', async ({ page }) => {
+    //1. Select the SPECIALTIES menu item in the navigation bar
+    await page.getByText('Specialties').click()
+    await page.waitForSelector('tbody')
```

Comment:
this wait is not needed, because on the next step the `click()` method will wait for "Add" button automatically